### ISSUE TITLE
setup.py logging in install_requires breaks install

### ISSUE THREAD

**Issue by drjsanger** [cancerit/archCompare#17]
https://github.com/cancerit/archCompare/blob/b5bf14ab2d9a824035402f2a94cba394cfc38ae7/setup.py#L14

When installing with python 3, `logging` is a core module. Including it in the `install_requires ` causes an external library to be installed which breaks on install:

```
$ pip install .
Processing /Users/drj/Documents/git/archCompare
Collecting logging (from archCompare==1.1.3)
  Using cached logging-0.4.9.6.tar.gz
    Complete output from command python setup.py egg_info:
    running egg_info
    creating pip-egg-info/logging.egg-info
    writing pip-egg-info/logging.egg-info/PKG-INFO
    writing dependency_links to pip-egg-info/logging.egg-info/dependency_links.txt
    writing top-level names to pip-egg-info/logging.egg-info/top_level.txt
    writing manifest file 'pip-egg-info/logging.egg-info/SOURCES.txt'
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/gz/7jz3f7d507sfh9r24qq9css0000b0m/T/pip-build-gxaszfe8/logging/setup.py", line 13, in <module>
        packages = ["logging"],
      File "/Users/drj/homebrew/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/Users/drj/homebrew/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/Users/drj/homebrew/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/setuptools/command/egg_info.py", line 278, in run
        self.find_sources()
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/setuptools/command/egg_info.py", line 293, in find_sources
        mm.run()
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/setuptools/command/egg_info.py", line 524, in run
        self.add_defaults()
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/setuptools/command/egg_info.py", line 560, in add_defaults
        sdist.add_defaults(self)
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/setuptools/command/py36compat.py", line 34, in add_defaults
        self._add_defaults_python()
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/setuptools/command/sdist.py", line 127, in _add_defaults_python
        build_py = self.get_finalized_command('build_py')
      File "/Users/drj/homebrew/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/cmd.py", line 298, in get_finalized_command
        cmd_obj = self.distribution.get_command_obj(command, create)
      File "/Users/drj/homebrew/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/dist.py", line 846, in get_command_obj
        klass = self.get_command_class(command)
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/setuptools/dist.py", line 582, in get_command_class
        self.cmdclass[command] = cmdclass = ep.load()
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2431, in load
        return self.resolve()
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2437, in resolve
        module = __import__(self.module_name, fromlist=['__name__'], level=0)
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/setuptools/command/build_py.py", line 15, in <module>
        from setuptools.lib2to3_ex import Mixin2to3
      File "/Users/drj/Documents/git/archCompare/test_env/lib/python3.6/site-packages/setuptools/lib2to3_ex.py", line 12, in <module>
        from lib2to3.refactor import RefactoringTool, get_fixers_from_package
      File "/Users/drj/homebrew/Cellar/python3/3.6.2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/lib2to3/refactor.py", line 17, in <module>
        import logging
      File "/private/var/folders/gz/7jz3f7d507sfh9r24qq9css0000b0m/T/pip-build-gxaszfe8/logging/logging/__init__.py", line 618
        raise NotImplementedError, 'emit must be implemented '\
                                 ^
    SyntaxError: invalid syntax
```

Removed logging from `install_requires` and I can install correctly:

```
$ pip install .
Processing /Users/drj/Documents/git/archCompare
Collecting beautifultable (from archCompare==1.1.3)
  Using cached beautifultable-0.4.0-py2.py3-none-any.whl
Installing collected packages: beautifultable, archCompare
  Running setup.py install for archCompare ... done
Successfully installed archCompare-1.1.3 beautifultable-0.4.0
```

**Comment by drjsanger** [comment]
See [feature/install_doc_fix](https://github.com/cancerit/archCompare/tree/feature/install_doc_fix) for current changes relating to this

**Comment by sb43** [comment]
Thanks for fixing, somehow it didn't picked by travis.
I will merge this branch to next hotfix release

**Comment by drjsanger** [comment]
I suspect because in `.travis.yml` it is not trying to install logging:

https://github.com/cancerit/archCompare/blob/344f557066365a85233a7d0e44ac28708fcb7be3/.travis.yml#L14-L18

**Comment by keiranmraine** [comment]
@sb43 your travis should be trying to do an install not installing the libraries separately.

e.g. `pip install .`

**Comment by drjsanger** [comment]
I will change this as part of the hotfix (I am aware I had a 'Friday moment') in starting a feature branch

**Comment by drjsanger** [comment]
Fixed in [1.1.4](https://github.com/cancerit/archCompare/tree/1.1.4)
